### PR TITLE
Add server warnings

### PR DESCRIPTION
**Pull request by lucekdudek**

### CODE REVIEW

File: ray_on_golem/server/services/ray.py [hunk 1]
```diff
@@ -248,16 +251,25 @@ async def _monitor_node(self, node_id: NodeId, activity: Activity) -> None:
...
                 msg = (
-                    f"Activity on {await self._golem_service.get_provider_desc(activity)} "
+                    f"Activity on {provider_desc} "
                     f"is no longer accessible. Terminating {node_id} {is_head_node=}"
                 )
                 logger.warning(msg)
                 logger.debug(msg, exc_info=True)
                 if is_head_node:
+                    self.add_service_warning(
+                        f"{datetime.now(timezone.utc).isoformat()} Terminating cluster as "
```
Comment (approxit) [inline ray_on_golem/server/services/ray.py:263]: Looks like the timestamp could be moved into `add_service_warning` as a bool with a default.

File: ray_on_golem/server/services/ray.py
[the remark this answers is not in the document]
Comment (lucekdudek) [reply ray_on_golem/server/services/ray.py:1]: Done

File: ray_on_golem/server/services/ray.py (hunk 1 above)
Comment (lucekdudek) [reply ray_on_golem/server/services/ray.py]: Done

File: ray_on_golem/server/services/ray.py [hunk 2]
```diff
@@ -70,23 +71,21 @@ def __init__(
...
         self._ssh_user: Optional[str] = None
+        super().__init__()
 
     async def shutdown(self) -> None:
         logger.info("Stopping RayService...")
         await self._stop_node_monitoring_tasks()
         await self._stop_head_node_to_webserver_tunnel()
         await self._stop_create_node_tasks()
         await self._destroy_nodes()
-        self._service_warnings.clear()
```
Comment (approxit) [inline ray_on_golem/server/services/ray.py]: This option would be useful in the mixin.

File: ray_on_golem/server/services/utils.py
Comment (approxit) [inline ray_on_golem/server/services/utils.py:1]: Would the `mixins.py` file name be more accurate in this case?

File: ray_on_golem/server/services/utils.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from datetime import datetime, timezone
+from typing import List
+
+
+class WarningMessagesMixin:
+    def __init__(self) -> None:
+        self._service_warnings = []
```
Comment (approxit) [inline ray_on_golem/server/services/utils.py:7]:
```suggestion
        self._warning_messagess = []
```
Maybe?

**Review by approxit — APPROVED** [review]

File: ray_on_golem/server/services/utils.py (hunk 3 above)
Comment (lucekdudek) [reply ray_on_golem/server/services/utils.py:7]: Yes

File: ray_on_golem/server/services/utils.py
Comment (lucekdudek) [reply ray_on_golem/server/services/utils.py:1]: Done

**Review by approxit — APPROVED** [review]

**Pull request merged**